_pytest/capture.py

```diff
@@ -252,6 +252,8 @@ def writelines(self, linelist):
         self.write(data)
 
     def __getattr__(self, name):
+        if name == 'name':
```

Comment:
Will do, but it is not really clear to me if this is the way to go after all, since it seems to be only necessary for logging.StreamHandler ?!

@RonnyPfannschmidt 
What API where you referring to that says this to be a string?